packages/checkout-ui-extensions/documentation/conditional-props.md

```diff
@@ -0,0 +1,88 @@
+
+```jsx
+<Button size={Style.default(`base`).when({hover: true}, `large`)}>
+  Click me
+</Button>
+```
+
+##### Multiple condition
+
+You can also provide multiple conditions to accomodate more complex interactions and requirements.
+
+```jsx
+<Button
+  size={Style.default(`base`)
+    .when({focus: true}, `large`)
+    .when({hover: true}, `large`)
+    .when({inlineViewportSize: `small`}, `extraLarge`)
+    .when({inlineViewportSize: `small`, focus: true}, `base`)}
+>
+  Click me
+</Button>
+```
+
+In this example, the following conditions are set:
+
+- Set the default size of the button to base
+- If button is focused, set the size to large
+- If button is hovered, set the size to large
+- If viewport's width is small, set the button's size to extraLarge
+- If viewport's width is small, and the focus is true, set the size to base
+
+### Style Helper class
+
+`Style` is a helper for authoring conditional values for prop styles.
+
+Write complex conditional styles based on one or more conditions (viewport sizes and interactive states) in a concise and expressive way.
+
+| Name    | Type                                                                                                                                                         | Description                                                                                                                                                                                                                                                                              |
+| ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| default | <code>(defaultValue: T) => <a href="#chainableconditionalstyle">ChainableConditionalStyle</a><<wbr>T<wbr>></code>                                            | Sets an optional `defaultValue` to use when no other condition is met.                                                                                                                                                                                                                   |
+| when    | <code>(conditions: <a href="#conditions">Conditions</a>, value: T) => <a href="#chainableconditionalstyle">ChainableConditionalStyle</a><<wbr>T<wbr>></code> | Adjusts the style based on different `conditions`. All `conditions`, expressed as a literal object, must be met for the associated `value` to be applied.<br /><br />The `when` method can be chained together to build more complex styles, and are not necessarily mutually exclusive. |
```

Comment:
`and are not necessarily mutually exclusive` is a bit confusing. What's the `are` referring to? I see a singular `method` and `styles`. Clause could use some cleanup when you're able

Reply:
I just dropped that part of the sentence, I don't think it was necessary.